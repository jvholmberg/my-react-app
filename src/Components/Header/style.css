h1.Header,
h2.Header,
h3.Header,
h4.Header,
h5.Header,
h6.Header {
  margin-bottom: .5rem;
  line-height: 1.2;
}

h1.Header {
  font-size: 2.5rem;
}
h2.Header {
  font-size: 2rem;
}
h3.Header {
  font-size: 1.75rem;
}
h4.Header {
  font-size: 1.5rem;
}
h5.Header {
  font-size: 1.25rem;
}
h6.Header {
  font-size: 1rem;
}