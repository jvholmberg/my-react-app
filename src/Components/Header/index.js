import React from 'react';

import './style.css';

/**
 * A simple header for page/section/content supports (h1-h6)
 * @param {boolean} [as='h1'] - The level of heading (h1-h6). Constants exist for this value
 * @returns {element} <Header>
 */
const Header = ({ h1, h2, h3, h4, h5, h6, className, children, ...props }) => {
  let cn = 'Header';
  if (className) cn = `${className} ${cn}`;

  if (h6) return (<h6 className={cn} {...props}>{children}</h6>);
  if (h5) return (<h5 className={cn} {...props}>{children}</h5>);
  if (h4) return (<h4 className={cn} {...props}>{children}</h4>);
  if (h3) return (<h3 className={cn} {...props}>{children}</h3>);
  if (h2) return (<h2 className={cn} {...props}>{children}</h2>);
  return (<h1 className={cn} {...props}>{children}</h1>);
}
export default Header;
