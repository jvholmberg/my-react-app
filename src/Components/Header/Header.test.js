import React from 'react';

import Header from './';

describe('Header', () => {
  it('H6 matches snapshot', () => {
    expect(shallow(
      <Header h6 />
    )).toMatchSnapshot();
  });
  it('H5 matches snapshot', () => {
    expect(shallow(
      <Header h5 />
    )).toMatchSnapshot();
  });
  it('H6 matches snapshot', () => {
    expect(shallow(
      <Header h4 />
    )).toMatchSnapshot();
  });
  it('H6 matches snapshot', () => {
    expect(shallow(
      <Header h3 />
    )).toMatchSnapshot();
  });
  it('H6 matches snapshot', () => {
    expect(shallow(
      <Header h2 />
    )).toMatchSnapshot();
  });
  it('H6 matches snapshot', () => {
    expect(shallow(
      <Header h1 />
    )).toMatchSnapshot();
  });
  it('Empty matches snapshot', () => {
    expect(shallow(
      <Header />
    )).toMatchSnapshot();
  });
  it('Empty with className matches snapshot', () => {
    expect(shallow(
      <Header className='className' />
    )).toMatchSnapshot();
  });
});
