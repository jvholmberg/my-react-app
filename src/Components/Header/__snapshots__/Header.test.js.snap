// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Empty matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Header",
    },
    "ref": null,
    "rendered": null,
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Header",
      },
      "ref": null,
      "rendered": null,
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header Empty with className matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    className="className"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "className Header",
    },
    "ref": null,
    "rendered": null,
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "className Header",
      },
      "ref": null,
      "rendered": null,
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header H5 matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    h5={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Header",
    },
    "ref": null,
    "rendered": null,
    "type": "h5",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Header",
      },
      "ref": null,
      "rendered": null,
      "type": "h5",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header H6 matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    h6={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Header",
    },
    "ref": null,
    "rendered": null,
    "type": "h6",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Header",
      },
      "ref": null,
      "rendered": null,
      "type": "h6",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header H6 matches snapshot 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    h4={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Header",
    },
    "ref": null,
    "rendered": null,
    "type": "h4",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Header",
      },
      "ref": null,
      "rendered": null,
      "type": "h4",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header H6 matches snapshot 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    h3={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Header",
    },
    "ref": null,
    "rendered": null,
    "type": "h3",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Header",
      },
      "ref": null,
      "rendered": null,
      "type": "h3",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header H6 matches snapshot 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    h2={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Header",
    },
    "ref": null,
    "rendered": null,
    "type": "h2",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Header",
      },
      "ref": null,
      "rendered": null,
      "type": "h2",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header H6 matches snapshot 5`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    h1={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Header",
    },
    "ref": null,
    "rendered": null,
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Header",
      },
      "ref": null,
      "rendered": null,
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
